apps/crn-server/scripts/opensearch/sync-opensearch-analytics.ts

```diff
@@ -151,46 +157,60 @@ export const exportAnalyticsData = async <T extends Metrics>(
-
-            do {
-              const teamProductivityResponse =
-                (await analyticsController.fetchTeamProductivity({
-                  take: PAGE_SIZE,
-                  skip: (teamPage - 1) * PAGE_SIZE,
-                  filter: { timeRange, outputType },
-                })) as ListResponse<MetricObject<T>>;
-
-              if (teamProductivityResponse) {
-                teamTotal = teamProductivityResponse.total;
-                teamPage++;
-                teamRecordCount += teamProductivityResponse.items.length;
-
-                const enrichedItems = teamProductivityResponse.items.map(
-                  (item) => ({
-                    ...item,
-                    timeRange,
-                    outputType,
-                  }),
-                );
+        // Helper to fetch all pages for a single timeRange/outputType combination
+        const fetchAllPagesForTeamCombination = async (
+          timeRange: (typeof timeRanges)[number],
+          outputType: (typeof outputTypes)[number],
+        ): Promise<MetricObject<T>[]> => {
+          const items: MetricObject<T>[] = [];
+          let page = 1;
+          let total = 0;
+          let recordCount = 0;
+
+          do {
+            const response = (await analyticsController.fetchTeamProductivity({
+              take: PAGE_SIZE,
+              skip: (page - 1) * PAGE_SIZE,
+              filter: { timeRange, outputType },
+            })) as ListResponse<MetricObject<T>>;
+
+            if (response) {
+              total = response.total;
+              page++;
+              recordCount += response.items.length;
+
+              const enrichedItems = response.items.map((item) => ({
+                ...item,
+                timeRange,
+                outputType,
+              }));
+
+              items.push(...enrichedItems);
+            }
+          } while (total > recordCount);
+
+          return items;
+        };
+
+        // Create all timeRange<->outputType combinations
+        const teamCombinations = timeRanges.flatMap((timeRange) =>
+          outputTypes.map((outputType) => ({ timeRange, outputType })),
+        );
 
-                allTeamProductivityItems.push(...enrichedItems);
-              }
-            } while (teamTotal > teamRecordCount);
-          }
-        }
-        /* eslint-enable no-await-in-loop */
+        // Process combinations with controlled concurrency
+        const teamResultArrays = await mapLimit(
+          teamCombinations,
+          MAX_CONCURRENT_COMBINATIONS,
+          async (combination: (typeof teamCombinations)[number]) =>
+            fetchAllPagesForTeamCombination(
+              combination.timeRange,
+              combination.outputType,
+            ),
+        );
 
         return {
-          total: allTeamProductivityItems.length,
-          items: allTeamProductivityItems,
+          total: teamResultArrays.flat().length,
+          items: teamResultArrays.flat(),
```

Comment:
calling .flat twice means going through the whole list and the lists inside of it twice better wrap everything like this:
```suggestion
        const teamResultArrays = (await mapLimit(
          teamCombinations,
          MAX_CONCURRENT_COMBINATIONS,
          async (combination: (typeof teamCombinations)[number]) =>
            fetchAllPagesForTeamCombination(
              combination.timeRange,
              combination.outputType,
            ),
        )).flat();
        
        return {
          total: teamResultArrays.length,
          items: teamResultArrays,
```